rewrite-java/src/main/java/org/openrewrite/java/search/SemanticallyEqual.java

```diff
@@ -1160,6 +1089,21 @@ public J.Switch visitSwitch(J.Switch _switch, J j) {
             return _switch;
         }
 
+        @Override
+        public J.SwitchExpression visitSwitchExpression(J.SwitchExpression _switch, J j) {
+            if (isEqual.get()) {
+                if (!(j instanceof J.SwitchExpression)) {
+                    isEqual.set(false);
+                    return _switch;
+                }
+
+                J.SwitchExpression compareTo = (J.SwitchExpression) j;
+                this.visit(_switch.getSelector(), compareTo.getSelector());
+                this.visit(_switch.getCases(), compareTo.getCases());
+            }
+            return _switch;
+        }
```

Comment:
This visit method was completely missing. It must have been forgotten when the method was added,

Reply:
Lots of things are missing from `SemanticallyEqual`, as it is a very broad concept. It could conceivably include considering `for` and `while` loops that implement the same behavior to be semantically equal. Bits and pieces have been implemented on an as-needed basis